### PR TITLE
[logging] UseOpenTelemetry extension & WithLogging default behavior

### LINKED ISSUE #4985: Default host behavior for OpenTelemetryBuilder With* extensions
Related to #4896

See: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4958#issuecomment-1776276235

Currently calling `WithMetrics` or `WithLogging` does NOT enable integration with the hosting pipeline (`ILoggerFactory` or `IMetricsListener`).

Based on discussions with @noahfalk & @samsp-msft it sounds like we want to enable those by default perhaps giving users a way to opt-out. Opening this issue to track that effort.

Updates...

Metrics: #4958 changes `WithMetrics` to register the `IMetricsListener`
Logging: TBD

### PR DESCRIPTION
**Pull request by CodeBlanch**
Follow-up to #4958
Fixes #4985
Relates to #4433

## Changes

* Adds `ILoggingBuilder.UseOpenTelemetry` experimental APIs. This brings full parity w.r.t. `IServiceCollection` / `IServiceProvider` that we have with metrics & tracing to logging.
* Updates `OpenTelemetryBuilder.WithLogging` experimental API to automatically register the `ILoggerProvider` integration. This mirrors what `WithMetrics` does (for `IMetricsListener` integration).

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [X] Unit tests added/updated
* [X] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry/CHANGELOG.md [hunk 1]
```diff
@@ -58,6 +58,12 @@
   implementationFactory)`.
   ([#4916](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4916))
 
+* Added the `ILoggingBuilder.UseOpenTelemetry` experimental API extension for
+  registering OpenTelemetry `ILogger` integration using `LoggerProviderBuilder`
+  which supports the full DI (`IServiceCollection` \ `IServiceProvider`) API
+  surface available already in tracing & metrics.
```
Comment (utpilla) [inline src/OpenTelemetry/CHANGELOG.md:64]:
```suggestion
  surface similar to tracing & metrics.
```

File: src/OpenTelemetry/Logs/ILogger/OpenTelemetryLoggingExtensions.cs [hunk 2]
```diff
@@ -48,6 +51,9 @@ public static class OpenTelemetryLoggingExtensions
     /// </remarks>
     /// <param name="builder">The <see cref="ILoggingBuilder"/> to use.</param>
     /// <returns>The supplied <see cref="ILoggingBuilder"/> for call chaining.</returns>
+#if EXPOSE_EXPERIMENTAL_FEATURES
+    [EditorBrowsable(EditorBrowsableState.Never)]
```
Comment (utpilla) [inline src/OpenTelemetry/Logs/ILogger/OpenTelemetryLoggingExtensions.cs:55]: We could add a comment here explaining why we are adding this attribute.

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry/CHANGELOG.md (hunk 1 above)
Comment (CodeBlanch) [reply src/OpenTelemetry/CHANGELOG.md]: Updated

File: src/OpenTelemetry/Logs/ILogger/OpenTelemetryLoggingExtensions.cs (hunk 2 above)
Comment (CodeBlanch) [reply src/OpenTelemetry/Logs/ILogger/OpenTelemetryLoggingExtensions.cs:57]: Updated

File: src/OpenTelemetry/Logs/ILogger/OpenTelemetryLoggingExtensions.cs [hunk 3]
```diff
@@ -75,19 +173,19 @@ public static class OpenTelemetryLoggingExtensions
...
-            // Logging:OpenTelemetry configuration.
-            services.Configure(configureOptions);
-        }
-
         // Note: This will bind logger options element (eg "Logging:OpenTelemetry") to OpenTelemetryLoggerOptions
         RegisterLoggerProviderOptions(services);
 
         services.AddOpenTelemetrySharedProviderBuilderServices();
 
+        if (configureOptions != null)
```
Comment (utpilla) [inline src/OpenTelemetry/Logs/ILogger/OpenTelemetryLoggingExtensions.cs:181]: Do we need a CHANGELOG entry for this change?

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry/Logs/ILogger/OpenTelemetryLoggingExtensions.cs (hunk 3 above)
Comment (CodeBlanch) [reply src/OpenTelemetry/Logs/ILogger/OpenTelemetryLoggingExtensions.cs:181]: Added

**Pull request merged**